enhancements/arbiter-clusters.md

```diff
@@ -0,0 +1,692 @@
+need small changes to support the arbiter node. The `StaticPod` controller and
+the `Guard` controller will need to be updated to add a helper method that
+allows CEO and other tools to easily turn on the `arbiter` node selector for
+their internal node listers.
+
+`ex:`
+
+```go
+
+func (c *Controller) WithArbiter() {
+    c.withArbiter = true;
+}
+...
+
+if c.withArbiter {
+    arbiterselector, err := labels.NewRequirement("node-role.kubernetes.io/arbiter", selection.Equals, []string{""})
+    arbiternodes, err := c.nodeLister.List(labels.NewSelector().Add(*arbiterselector))
+    nodes = append(nodes, arbiternodes...)
+}
+...
+
+```
+
+### Authentication Operator Change
+
+The authentication operator runs a [readiness check
+controller](https://github.com/openshift/cluster-authentication-operator/blob/2d71f164af3f3e9c84eb40669d330df2871956f5/pkg/controllers/readiness/unsupported_override.go#L58-L70)
+that enforces for all non `SingleReplicaTopologyMode` topologies or an
+explicitly override, there must be 3 or more `kube-apiservers`. We need to add
+another check to this switch statement to allow minimum 2 replicas for
+`HighlyAvailableArbiter` topologies.
+
+### Hosted Control Plane Change
+
+We will not support running this type of topology for hosted control plane
+installations. We need to update HyperShift to check for the
+`HighlyAvailableArbiter` control plane topology and prevent HCP installations on
+Arbiter clusters. This will need to be done at the `hypershift` cli level as
+well as any bootstrap component that can be created outside of the CLI flow.
+
+### Alternative Install FLow Change
+
+We currently have a few different options for different needs when installing
+OCP that need to also be updated. Work done in the installer should be reflected
+on the Assisted Installer Service, Agent Based Installer and ZTP.
+
+In ABI (Agent Based Installer) the bootstrap node, which runs the
+`assisted-service` as well as the `in-place bootstrap`, is required to be a
+control plane node. This should not run on the arbiter node since it will have
+less compute and thus we would prefer it to run on one of the larger control
+plane nodes.
+
+In both ABI and Assisted, the node that runs the `in-place bootstrap` is the
+last to be rebooted to join the cluster. We would require this to be the one of
+the large control plane node, and since 2 etcd replicas are all that is required
+to add the third node safely, a 1 control plane + 1 arbiter cluster satisfies
+this requirement. This also would mean less major code changes.
+
+### OLM Filter Addition
+
+With this change it would be prudent to add the ability to allow operators to
+specify which control plane topology they support. This gives us more guards
+against installing layered components on unsupported or unconsidered topologies
+like the Master+Arbiter in this enhancement.
+
+This is a nice to have before we go GA and we should be looking into adding it
+to OLM.
```

Comment:
+1. The OLM team has some thoughts related to this along the lines of generic cluster-provided properties and layered-product provided constraints. This seems to be one use case for that feature. We'll have to think carefully about implementing this in a more general "properties vs. constraints" kind of design.

I don't know that I would make your GA contingent on this feature though since OCP already has multiple topology variants and OLM does not have this feature yet.

One thing we _could_ consider in the meantime is a hint annotation in the CSV that the OCP console could be made aware of. OCP console already supports some of these annotations, and uses them to filter views presented to users.

Reply:
Awesome, I have it as a nice to have for GA, so we wont let be a critical part of our checklist. Thanks for your input Joe!